### ISSUE TITLE
Bug - Notification Drawer should allow focus trap

### ISSUE THREAD

**Issue by logonoff** [patternfly/patternfly-react#11703]
**Describe the problem**
A clear and concise description of the problem. Which pages are affected?

When using a keyboard to navigate the page in https://www.patternfly.org/components/notification-drawer/react-demos/basic/, keyboard focus is given to the first button or link in the drawer, which is correct. 

However, when a user goes and closes the notification drawer, keyboard focus does not return to the notification button, and seems to return to the browser chrome, which is disorienting. 

Moreover, after tabbing to the last item in the notification drawer, the next tab press should loop back to the first item in the notification drawer; however, this is not done.

**Expected behavior**
A clear and concise description of the expected behavior.

In the demo, keyboard focus should be returned to the notification badge after the notification drawer is closed, and the keyboard focus should be trapped entirely within the notification drawer when it is open.

**Screenshots**
If applicable, add screenshots to help explain the issue.

**Any other information?**

See this blog post:

https://codepen.io/unleashalicia/pen/YzXrJBN
https://knowbility.org/blog/2020/accessible-slide-menus

**Comment by logonoff** [comment]
activity

**Comment by logonoff** [comment]
Bump
